src/iosurface.cs

```diff
@@ -0,0 +1,220 @@
+		NSString PixelFormatKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPixelSizeCastingAllowed")]
+		NSString PixelSizeCastingAllowedKey { get; } 
+	}
+
+	[StrongDictionary ("IOSurfacePropertyKey")]
+	partial interface IOSurfaceOptions {
+		nint AllocSize { get; set; }
+		nint Width { get; set; }
+		nint Height { get; set; }
+		nint BytesPerRow { get; set; }
+		nint BytesPerElement { get; set; }
+		nint ElementWidth { get; set; }
+		nint ElementHeight { get; set; }
+		nint Offset { get; set; }
+		NSDictionary [] PlaneInfo { get; set; }
+		nint PlaneWidth { get; set; }
+		nint PlaneHeight { get; set; }
+		nint PlaneBytesPerRow { get; set; }
+		nint PlaneOffset { get; set; }
+		nint PlaneSize { get; set; }
+		nint PlaneBase { get; set; }
+		nint PlaneBytesPerElement { get; set; }
+		nint PlaneElementWidth { get; set; }
+		nint PlaneElementHeight { get; set; }
+		IOSurfaceMemoryMap CacheMode { get; set; }
+		uint PixelFormat { get; set; }
+		bool PixelSizeCastingAllowed { get; set; }
+	}
+
+	// @interface IOSurface : NSObject <NSSecureCoding>
+	[iOS (10,0), Mac(10,12)]
+	[BaseType (typeof(NSObject))]
+	interface IOSurface : NSSecureCoding
+	{
+		[Internal, Export ("initWithProperties:")]
+		IntPtr Constructor (NSDictionary properties);
+
+		[Wrap ("this (properties == null ? null : properties.Dictionary)")]
+		IntPtr Constructor (IOSurfaceOptions properties);
+	
+		[Internal, Export ("lockWithOptions:seed:")]
+		int _Lock (IOSurfaceLockOptions options, IntPtr seedPtr);
+	
+		[Internal, Export ("unlockWithOptions:seed:")]
+		int _Unlock (IOSurfaceLockOptions options, IntPtr seed);
+	
+		[Export ("allocationSize")]
+		nint AllocationSize { get; }
+	
+		[Export ("width")]
+		nint Width { get; }
+	
+		[Export ("height")]
+		nint Height { get; }
+	
+		[Export ("baseAddress")]
+		IntPtr BaseAddress { get; }
+	
+		[Export ("pixelFormat")]
+		uint PixelFormat { get; }
+	
+		[Export ("bytesPerRow")]
+		nint BytesPerRow { get; }
+	
+		[Export ("bytesPerElement")]
+		nint BytesPerElement { get; }
+	
+		[Export ("elementWidth")]
+		nint ElementWidth { get; }
+	
+		[Export ("elementHeight")]
+		nint ElementHeight { get; }
+	
+		[Export ("seed")]
+		uint Seed { get; }
+	
+		[Export ("planeCount")]
+		nuint PlaneCount { get; }
+	
+		[Export ("widthOfPlaneAtIndex:")]
+		nint GetWidth (nuint planeIndex);
+	
+		[Export ("heightOfPlaneAtIndex:")]
+		nint GetHeight (nuint planeIndex);
+	
+		[Export ("bytesPerRowOfPlaneAtIndex:")]
+		nint GetBytesPerRow (nuint planeIndex);
+	
+		[Export ("bytesPerElementOfPlaneAtIndex:")]
+		nint GetBytesPerElement (nuint planeIndex);
+	
+		[Export ("elementWidthOfPlaneAtIndex:")]
+		nint GetElementWidth (nuint planeIndex);
+	
+		[Export ("elementHeightOfPlaneAtIndex:")]
+		nint GetElementHeight (nuint planeIndex);
+	
+		[Export ("baseAddressOfPlaneAtIndex:")]
+		IntPtr GetBaseAddress (nuint planeIndex);
+	
+		[Export ("setAttachment:forKey:")]
+		void SetAttachment (NSObject anObject, NSString key);
```

Comment:
It should not be a `string`, the value is used as a pointer/token, not really as a user string.